src/core/mapping/base_mapper.cc

```diff
@@ -16,27 +16,65 @@
 
 #include <cstdlib>
 #include <sstream>
+#include <unordered_map>
 
 #include "legion/legion_mapping.h"
 #include "mappers/mapping_utilities.h"
 
 #include "core/data/store.h"
 #include "core/mapping/base_mapper.h"
 #include "core/mapping/instance_manager.h"
-#include "core/mapping/task.h"
+#include "core/mapping/operation.h"
 #include "core/runtime/projection.h"
 #include "core/runtime/shard.h"
 #include "core/utilities/linearize.h"
 #include "legate_defines.h"
 
 using LegionTask = Legion::Task;
+using LegionCopy = Legion::Copy;
 
 using namespace Legion;
 using namespace Legion::Mapping;
 
 namespace legate {
 namespace mapping {
 
+namespace {
+
+const std::vector<StoreTarget>& default_store_targets(Processor::Kind kind)
+{
+  static const std::map<Processor::Kind, std::vector<StoreTarget>> defaults = {
+    {Processor::LOC_PROC, {StoreTarget::SYSMEM}},
+    {Processor::TOC_PROC, {StoreTarget::FBMEM, StoreTarget::ZCMEM}},
+    {Processor::OMP_PROC, {StoreTarget::SOCKETMEM, StoreTarget::SYSMEM}},
```

Comment:
You could also close out https://github.com/nv-legate/legate.core/issues/335 by returning `SOCKETMEM` only if there are actually socket memories present on the machine.

Reply:
I feel that shouldn't be necessary, because returning `SOCKETMEM` has the same effect as returning `SYSMEM` when there is no socket memory thanks to this line: https://github.com/nv-legate/legate.core/blob/b5ce4428719a41c1113ec536a63acf68767fa9fd/src/core/mapping/base_mapper.cc#L108 can you remind me exactly what was causing #335? that might need a separate fix.

Reply:
The issue describes it, and has a link to the changed line. I dont remember the cause — at the time, you just told me to make that change and the problem went away.

Reply:
dug old conversations and found this comment:

> so that basically means that the map_copy tries to find source and/or target instances in the system memory

so the issue sounds like that `map_copy` was implemented poorly. this PR should fix it as well, so hopefully it will also fix #335 with no extra change.